lib/ansible/modules/cloud/hosttech/hosttech_dns_record.py

```diff
@@ -0,0 +1,1192 @@
+        command.add_simple_command('changeTTL', ip=ip, ttl=ttl)
+        try:
+            return self._execute(command, 'changeTTLResponse', int)
+        except WSDLError as e:
+            # FIXME
+            raise
+
+
+# ##########################################################################################################
+# # Ansible module #########################################################################################
+# ##########################################################################################################
+
+
+def run_module():
+    module_args = dict(
+        state=dict(type='str', choices=['present', 'absent', 'get'], required=True),
+        zone=dict(type='str', required=True),
+        record=dict(type='str', required=True),
+        ttl=dict(required=False, type='int', default=3600),
+        type=dict(choices=['A', 'CNAME', 'MX', 'AAAA', 'TXT', 'PTR', 'SRV', 'SPF', 'NS', 'CAA'], required=True),
+        value=dict(required=False, type='list', default=[]),
+        overwrite=dict(required=False, default=False, type='bool'),
+        hosttech_username=dict(type='str', required=True),
+        hosttech_password=dict(type='str', required=True, no_log=True),
+    )
+    required_if = [('state', 'present', ['value']), ('state', 'absent', ['value'])]
+    module = AnsibleModule(argument_spec=module_args, required_if=required_if, supports_check_mode=True)
+
+    if not HAS_LXML_ETREE:
+        module.fail_json(msg='Needs lxml Python module (pip install lxml)')
+
+    # Get zone and record.
+    zone_in = module.params.get('zone').lower()
+    record_in = module.params.get('record').lower()
+    if zone_in[-1:] == '.':
+        zone_in = zone_in[:-1]
+    if record_in[-1:] == '.':
+        record_in = record_in[:-1]
+
+    # Convert record to prefix
+    if not record_in.endswith('.' + zone_in) and record_in != zone_in:
+        module.fail_json(msg='Record must be in zone')
+    if record_in == zone_in:
+        prefix = None
+    else:
+        prefix = record_in[:len(record_in) - len(zone_in) - 1]
+
+    # Create API and get zone information
+    api = HostTechAPI(module.params.get('hosttech_username'), module.params.get('hosttech_password'), debug=False)
+    try:
+        zone = api.get_zone(zone_in)
+        if zone is None:
+            module.fail_json(msg='Zone not found')
+    except HostTechAPIAuthError as e:
+        module.fail_json(msg='Cannot authenticate', error=e.message)
+    except HostTechAPIError as e:
+        module.fail_json(msg='Internal error (API level)', error=e.message)
+    except WSDLException as e:
+        module.fail_json(msg='Internal error (WSDL level)', error=e.message)
+
+    # Find matching records
+    type_in = module.params.get('type')
+    records = []
+    for record in zone.records:
+        if record.prefix == prefix and record.type == type_in:
+            records.append(record)
+
+    # Process get
+    if module.params.get('state') == 'get':
```

Comment:
By policy, mixing a get or list function into a "normal" module is not accepted. A list or get function should be turned into a proper facts module.

Reply:
Ok, makes sense. I guess the `route53` module has been like that longer than this policy. I'll split the module up into two, then. Is it OK if the second module is in this PR as well, or should it be a separate PR?

Reply:
Yes, this policy is younger than route53.
Each module has to be in a separate pr, however this is sometimes difficult if you also have shared code for module_utils.

Reply:
There will be some shared code, since most of the API functionality will be needed by both modules. I'll split this up first, and then we can decide later whether to move the fact gathering module to another (later) PR or not.

Reply:
Ok, I split it up. Now ansibot complains about two modules in the PR, as (almost) expected ;)
There's another thing I noticed: could it be that the module should better be in the net_tools category (https://docs.ansible.com/ansible/latest/list_of_net_tools_modules.html)? If yes: should I move it around in this PR, or close this PR and open a new one?